### PR TITLE
(Documentation) Update from the new Engine API and plugin approach

### PR DESCRIPTION
**Pull request by tsunammis**
This pull-request integrate the new documentation which features the new plugin approach of tartiflette.

The tutorial project will be also updated -> https://github.com/tartiflette/tartiflette-aiohttp-tutorial/pull/3

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.yml]
+indent_size = 2
+trim_trailing_whitespace = false
+
+[Jenkinsfile]
```
Comment (abusi) [inline .editorconfig]: No jenkins in tartiflette ;)

File: docs/tutorial/rate-limit-fields-with-directives.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ from typing import Any, Callable, Dict, Optional
 
 import time
 
-from tartiflette import Directive
+from tartiflette.directive import Directive
```
Comment (abusi) [inline docs/tutorial/rate-limit-fields-with-directives.md]: Directive is importable directly from tartiflette

File: docs/tutorial/dynamic-introspection.md [hunk 3]
```diff
@@ -87,8 +112,8 @@ $ python -m recipes_manager
 
 Execute the following query to retrieve the list of mutation fields.
 
-* Without the `@nonIntrospectable` you **will** see the `updateRecipe` field.
-* With the `@nonIntrospectable` in your schema you **won't** see the `updateRecipe` field.
+* Without the `@auth(role: "admin")` you **will** see the `updateRecipe` field.
+* With the `@auth(role: "admin") in your schema you **won't** see the `updateRecipe` if your are't using the `localhost:8080` endpoint.
```
Comment (Maximilien-R) [inline docs/tutorial/dynamic-introspection.md]:
```suggestion
* With the `@auth(role: "admin") in your schema you **won't** see the `updateRecipe` if your aren't using the `localhost:8080` endpoint.
```

File: docs/tutorial/dynamic-introspection.md [hunk 4]
```diff
@@ -19,32 +19,57 @@ You will be able to:
...
 
-## **recipes_manager/directives/non_introspectable.py**
+## **recipes_manager/directives/auth.py**
 
-The idea of the following code is to create a new directive called "@nonIntrospectable" which allow you to hide some fields and/or types from the introspection query.
+The idea of the following code is to create a new directive called "@auth" which allow you to:
+* hide fields and/or types from the introspection query.
+* restrict the access to fields and/or types
+
+In this following code, we check that the user uses the API endpoint `localhost:8080`, if the user doesn't use this one, like 127.0.0.1:8080 or another, an error will be thrown.
```
Comment (Maximilien-R) [inline docs/tutorial/dynamic-introspection.md]:
```suggestion
In this following code, we check that the user uses the API endpoint `localhost:8080`, if the user doesn't use this one, like `127.0.0.1:8080` or another, an error will be thrown.
```

File: docs/tutorial/dynamic-introspection.md [hunk 5]
```diff
@@ -19,32 +19,57 @@ You will be able to:
...
+
+Useless to say that this example is suited for this tutorial and must not be put in a real application.
 
 ```python
 from typing import Any, Callable, Dict, Optional
 
-from tartiflette import Directive
+from tartiflette.directive import Directive
```
Comment (Maximilien-R) [inline docs/tutorial/dynamic-introspection.md]: > Directive is importable directly from tartiflette - cc. @abusi

**Pull request merged**